src/ZuluSCSI_disk.cpp

```diff
@@ -500,21 +540,160 @@ static void scsiDiskLoadConfig(int target_idx, const char *section)
     memset(tmp, 0, sizeof(tmp));
     ini_gets(section, "Serial", "", tmp, sizeof(tmp), CONFIGFILE);
     if (tmp[0]) memcpy(img.serial, tmp, sizeof(img.serial));
+
+    if (strlen(section) == 5 && strncmp(section, "SCSI", 4) == 0) // allow within target [SCSIx] blocks only
+    {
+        ini_gets(section, "ImgDir", "", tmp, sizeof(tmp), CONFIGFILE);
+        if (tmp[0])
+        {
+            logmsg("-- SCSI", target_idx, " using image directory \'", tmp, "'");
+            img.image_directory = true;
+        }
+    }
 }
 
-// Check if image file name is overridden in config
-bool scsiDiskGetImageNameFromConfig(image_config_t &img, char *buf, size_t buflen)
+int scsiDiskGetNextImageName(image_config_t &img, char *buf, size_t buflen)
 {
     int target_idx = img.scsiId & 7;
 
     char section[6] = "SCSI0";
     section[4] = '0' + target_idx;
 
-    char key[5] = "IMG0";
-    key[3] = '0' + img.image_index;
+    // sanity check: is provided buffer is long enough to store a filename?
+    if (buflen < MAX_FILE_PATH)
+    {
+        panic("scsiDiskGetNextImageName called with illegal buflen");
+    }
+
+    if (img.image_directory)
+    {
+        // is this needed to prevent an inadvertent callback during all these opens?
```

Comment:
No, it is not needed. The callback should nulled out after use by any function that sets it. It also only triggers if the data buffer pointer passed to `file.read()` matches the one passed to `set_sd_callback()`.

Reply:
Perfect, removed.